### PR TITLE
feat(plugin-text-classification): TextClassifier implemented #BOT-349

### PR DESCRIPTION
**Pull request by elozano98**
**Depends on #1475 .** Please, review it first. ⚠️ 

## Description
The `TextClassifier` class is responsible for classifying a given text. 

## Approach taken / Explain the design
This class takes the given text and:
1. Processes it to get the input tensor.
2. Uses the model to predict the output.
3. Processes the output so it returns a list with all possible intents and their corresponding confidences.

## Testing

The pull request...

- [ ] has unit tests
- [ ] has integration tests

### CODE REVIEW

File: packages/botonic-plugin-text-classification/src/model/text-classifier.ts [hunk 1]
```diff
@@ -1,14 +1,34 @@
...
-import { TextClassificationConfig } from '@botonic/nlp/lib/tasks/text-classification/storage/types'
+import type { TextClassificationConfig } from '@botonic/nlp/lib/tasks/text-classification/storage/types'
+import type { Tensor3D } from '@tensorflow/tfjs'
 import { LayersModel } from '@tensorflow/tfjs'
 
+import { PredictionProcessor } from '../process/prediction-processor'
+import { TextProcessor } from '../process/text-processor'
+
+// @ts-ignore
+global.fetch = fetch
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/src/model/text-classifier.ts:11]: where do you use fetch?
Your link says it's a dirty hack we shoulnt use in production https://stackoverflow.com/a/51466112/145289
remove both usages if not required. if required, try to encapsulate in  a function or use a 3rd party package as suggested in  stackoverflow

**Review by dpinol — CHANGES_REQUESTED** [review]

File: packages/botonic-plugin-text-classification/src/model/text-classifier.ts (hunk 1 above)
Comment (vanbasten17) [reply packages/botonic-plugin-text-classification/src/model/text-classifier.ts:11]: @dpinol , the reason of this is that `tfjs` uses fetch internally to load a model.json given a url, so the package will work correctly in browser envs but not in lambda (node). In the same thread there are some solutions like `node-fetch` or `cross-fetch`. We should give them a try :)

**Review by vanbasten17 — CHANGES_REQUESTED** [review]
Let's solve the non-usage in production of `global.fetch` with a more suitable solution, do some tests by deploying the bot and then we can go for it.

File: packages/botonic-plugin-text-classification/src/model/text-classifier.ts (hunk 1 above)
Comment (dpinol) [reply packages/botonic-plugin-text-classification/src/model/text-classifier.ts:11]: @elozano98 can you create a ticket to solve this in a more elegant way?
Please add st like this:

> Your link says it's a dirty hack we shoulnt use in production https://stackoverflow.com/a/51466112/145289
> remove both usages if not required. if required, try to encapsulate in a function or use a 3rd party package as suggested in stackoverflow

Comment (elozano98) [reply packages/botonic-plugin-text-classification/src/model/text-classifier.ts:11]: I have created [a ticket](https://linear.app/hubtype/issue/BOT-357/remove-fetch-from-nlp-plugins) for this (will be implemented in another PR).

Comment (elozano98) [reply packages/botonic-plugin-text-classification/src/model/text-classifier.ts]: Fixed, cross-fetch used! @dpinol

**Review by vanbasten17 — APPROVED** [review]

File: packages/botonic-plugin-text-classification/src/model/model-info.ts [hunk 2]
```diff
@@ -11,7 +12,10 @@ export class ModelInfo {
 
   constructor(readonly locale: Locale, readonly uri: string) {
     this.config = axios({ url: `${this.uri}/${CONFIG_FILENAME}` })
-    this.model = loadLayersModel(`${this.uri}/${MODEL_FILENAME}`)
+    // fetchFunc is defined to replace window.fetch to work in node environment
```
Comment (dpinol) [inline packages/botonic-plugin-text-classification/src/model/model-info.ts:15]: please create a ticket to unify both ModelInfo classes.
 
* create a single class template at botonic-nlp.  Just modify this
```
 export class GenericModelInfo<T> {
  private readonly config: AxiosPromise<T>
async getConfig(): Promise<T> {
```
* Move the constants CONFIG_FILENAME & MODEL_FILENAME to be static fields of GenericModelInfo. constants should be as close as possible to the class that use them. They only need to be in a centralized placed when they need to be configured in a centralized way through env vars. even in that case, there should be a class which encapsulates them duplicating them and encapsulating them
* at each ner & textclassifer, 
export class NerModelInfo extends GenericModelInfo<NerConfig> {}

**Review by dpinol — APPROVED** [review]

**Pull request merged**